spark3-extensions/src/main/scala/org/apache/spark/sql/catalyst/expressions/AccumulateFiles.scala

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.spark.sql.catalyst.expressions
+
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.expressions.codegen.CodegenFallback
+import org.apache.spark.sql.catalyst.utils.SetAccumulator
+import org.apache.spark.sql.types.{DataType, IntegerType}
+
+case class AccumulateFiles(
+    filesAccumulator: SetAccumulator[String],
+    child: Expression) extends UnaryExpression with CodegenFallback {
+  override def dataType: DataType = IntegerType
+
+  override def nullable: Boolean = true
+  override def prettyName: String = "AccumulateFiles"
+  override lazy val deterministic = false
```

Comment:
Why isn't this deterministic? Are you setting this to avoid pushdown or other optimization?

Reply:
@rdblue Right Ryan. It's probably not needed. Since this is updating an accumulator and in a way has a state to it, i felt its better to make is non-deterministic.